projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -333,74 +312,22 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * Private method to update the visible previews navigating to the left (previous).
    */
   private previous(): void {
-    if (!this.images) {
+    if (!this.images || !this.previewConfig) {
       throw new Error('Internal library error - images must be defined');
     }
-    // check if prevImage should be blocked
-    if (this.isPreventSliding(0)) {
```

Comment:
preventSliding depends whether next() or previous() is called, so I moved the isPreventSliding logic within these two methods for better contextualization.